### ISSUE TITLE
Support 'type statement'

### ISSUE THREAD

**Issue by TakWolf** [BrianPugh/cyclopts#190]
https://docs.python.org/3.12/reference/simple_stmts.html#type
https://docs.python.org/3.12/library/typing.html#typing.TypeAlias

```python
from typing import TypeAlias, Literal
from cyclopts import App
from tools import configs

app = App()

FontSize: TypeAlias = Literal[10, 12, 16]
type BoxSize = Literal[10, 12, 16]

@app.default
def main(
        font_size: FontSize,
        box_size: BoxSize,
):
    print()

if __name__ == '__main__':
    app()

```

```text
(.venv) takwolf@bogon ark-pixel-font % python -m tools.cli -h                          
Usage: main COMMAND [ARGS] [OPTIONS]

╭─ Commands ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --help,-h  Display this message and exit.                                                                                                                                                                                                                                                                            │
│ --version  Display application version.                                                                                                                                                                                                                                                                              │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Parameters ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ *  FONT-SIZE,--font-size  [choices: 10,12,16] [required]                                                                                                                                                                                                                                                             │
│ *  BOX-SIZE,--box-size    [required]                                                                                                                                                                                                                                                                                 │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
(.venv) takwolf@bogon ark-pixel-font % python -m tools.cli --font-size 10 --box-size 10
╭─ Error ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Invalid value for "--box-size". 'typing.TypeAliasType' object is not callable                                                                                                                                                                                                                                        │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

```

'type' is a new feature added in python 3.12, Unlike 'TypeAlias', 'type' (TypeAliasType) is lazy in evaluating.

About TypeAlias in python document: https://docs.python.org/3.12/library/typing.html#typing.TypeAlias


> Deprecated since version 3.12: [TypeAlias](https://docs.python.org/3.12/library/typing.html#typing.TypeAlias) is deprecated in favor of the [type](https://docs.python.org/3.12/reference/simple_stmts.html#type) statement, which creates instances of [TypeAliasType](https://docs.python.org/3.12/library/typing.html#typing.TypeAliasType) and which natively supports forward references. Note that while [TypeAlias](https://docs.python.org/3.12/library/typing.html#typing.TypeAlias) and [TypeAliasType](https://docs.python.org/3.12/library/typing.html#typing.TypeAliasType) serve similar purposes and have similar names, they are distinct and the latter is not the type of the former. Removal of [TypeAlias](https://docs.python.org/3.12/library/typing.html#typing.TypeAlias) is not currently planned, but users are encouraged to migrate to [type](https://docs.python.org/3.12/reference/simple_stmts.html#type) statements.

I think he should be supported in the future

**Comment by BrianPugh** [comment]
Thanks for opening the issue! This is now supported in v2.9.0

**Comment by BrianPugh** [comment]
I forgot to update how `--help` resolves these for choces; this will be resolved in v2.9.1 in just a sec.

**Comment by TakWolf** [comment]
`--help` seems that there are still issues in v2.9.1

```python
from typing import Literal, TypeAlias

from cyclopts import App

app = App()

FontSingleFormat: TypeAlias = Literal['otf', 'woff2', 'ttf', 'bdf', 'pcf']
FontCollectionFormat: TypeAlias = Literal['otc', 'ttc']


@app.default
def main(
        font_formats: list[FontSingleFormat | FontCollectionFormat] | None = None
):
    print(font_formats)


if __name__ == '__main__':
    app()
```

```text
(.venv) % python main.py -h
Usage: main COMMAND [ARGS] [OPTIONS]

╭─ Commands ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --help,-h  Display this message and exit.                                                                                                                                                                                                                                                                            │
│ --version  Display application version.                                                                                                                                                                                                                                                                              │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Parameters ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ FONT-FORMATS,--font-formats,--empty-font-formats  [choices: otf,woff2,ttf,bdf,pcf,otc,ttc]                                                                                                                                                                                                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

```python
from typing import Literal

from cyclopts import App

app = App()

type FontSingleFormat = Literal['otf', 'woff2', 'ttf', 'bdf', 'pcf']
type FontCollectionFormat = Literal['otc', 'ttc']


@app.default
def main(
        font_formats: list[FontSingleFormat | FontCollectionFormat] | None = None
):
    print(font_formats)


if __name__ == '__main__':
    app()
```

```text
(.venv) % python main.py -h
Usage: main COMMAND [ARGS] [OPTIONS]

╭─ Commands ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --help,-h  Display this message and exit.                                                                                                                                                                                                                                                                            │
│ --version  Display application version.                                                                                                                                                                                                                                                                              │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Parameters ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ FONT-FORMATS,--font-formats,--empty-font-formats                                                                                                                                                                                                                                                                     │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

**Comment by BrianPugh** [comment]
Thanks for the additional report! This is due to some inconsistencies in how python internally defines unions. I have a more comprehensive union-check, but I forgot to use it when getting help options! Fix incoming and I'll release 2.9.2.
